extra/bunotel/option.go

```diff
@@ -30,3 +32,28 @@ func WithFormattedQueries(format bool) Option {
 		h.formatQueries = format
 	}
 }
+
+// WithTracerProvider returns an Option to use the TracerProvider when
+// creating a Tracer.
+func WithTracerProvider(tp trace.TracerProvider) Option {
+	return func(h *QueryHook) {
+		if tp != nil {
+			h.tracer = tp.Tracer("github.com/uptrace/bun")
```

Comment:
It is a bit strange that you pass `trace.TracerProvider` just to create a tracer. Also, what if you want to change tracer name?

Can't we instead accept a `tracer`?

The same applies to `WithMeterProvider`. Should not we accept a `meter`?

Reply:
I considered directly configuring the tracer and meter through options instead of TracerProvider and MeterProvider. However, I decided to do it this way because in the future, there might be a need to create different tracers and meters within the QueryHook. The current approach ensures compatibility in the future without breaking any existing functionality.
Thanks.

Reply:
Moreover, I think it is advisable to avoid allowing users to customize the name of tracer and meter to prevent ambiguity in the otel information.